ocw/lib/cleanup.py

```diff
@@ -47,6 +47,23 @@ def list_clusters():
                     type(e).__name__, vault_namespace), traceback.format_exc())
 
 
+def delete_snapshots():
+    cfg = ConfigFile()
+    if cfg.has('snapshots'):
+        for vault_namespace in cfg.getList(['snapshots', 'namespaces'], ['']):
+            try:
+                delete_after_days = date.today() - timedelta(days=cfg.getInt(['snapshots', 'delete_after_days'], 30))
```

Comment:
first of all no need to apologies in PRs - I always keep saying that review process does not have any requirement on expertise in certain project or even technology . You are more than welcome to ask **all** your question and share **all** of 
 your suggestions . I just can not give you any guaranties that everything will be applied :) 
 
Now regarding your certain suggestion here - first of all I just follow the pattern which already existing in `ConfigFile` class . Which already have same type specific getters for List and Boolean . So I would say even if follow your suggestion I would prefer to do it outside this PR and also change existing getters for Boolean and List. Can not judge how much pythonic this approach is in comparison to one which you suggesting ... to sum up - I can not say that I am against your idea but I would like to see some code . At least small code snippet here in this thread at best separate PR for refactor ConfigFIle class

Reply:
so either
`days=int(cfg.get(['snapshots', 'delete_after_days'], 30))`
or create a decorator (keep in mind that i have not dig to the code to deep, so i provide what i had in mind with n example which i run on my interpeter)
```
In [19]: def decor_int(f): 
    ...:     def wrapper(*args, **kwargs): 
    ...:         args = map(int, args) 
    ...:         return f(*args, **kwargs) 
    ...:     return wrapper 
    ...:                                                                                                                                                                                                                                     

In [20]: @decor_int 
    ...: def foo(a): 
    ...:     return a 
    ...:                                                                                                                                                                                                                                     

In [21]: foo('1')                                                                                                                                                                                                                            
Out[21]: 1
```

Reply:
look at this https://docs.python.org/3/library/configparser.html#configparser.ConfigParser.getint
it is the python default lib...
If python would have something like context related return value, then I would go with something implicit, but here I think it is ok.

I don't understand the decorator idea.
We have a function `get()` which return a string by default.
If I now would do a 
```
@decor_int
def get()
```
this function would always return a `int`, or?

Reply:
regarding the decorator i was thinking to put some logic there so for specific variables it will return int. not for all. but the point is not to use the decorator but it looks strange to have a wrapper for int(). it is more clear and 'pure' if in this particular case you have `days=int(cfg.get(['snapshots', 'delete_after_days'], 30))`.

Reply:
> regarding the decorator i was thinking to put some logic there so for specific variables it will return int. not for all. but the point is not to use the decorator but it looks strange to have a wrapper for int(). it is more clear and 'pure' if in this particular case you have `days=int(cfg.get(['snapshots', 'delete_after_days'], 30))`.

For me personally , I find it more cleaner to have dedicated method `getInt` than doing same transformation all over the code where we need int from this getter .